scipy/stats/stats.py

```diff
@@ -3231,23 +3234,47 @@ def f_oneway(*args):
     return F_onewayResult(f, prob)
 
 
+class PearsonRConstantInputWarning(RuntimeWarning):
+    """
+    Warning generated by `pearsonr` when an input is constant.
+    """
+
+    def __init__(self, msg=None):
+        if msg is None:
+            msg = ("An input array is constant; the correlation coefficent "
+                   "is not defined.")
+        self.args = (msg,)
+
+
+class PearsonRNearConstantInputWarning(RuntimeWarning):
+    """
+    Warning generated by `pearsonr` when an input is nearly constant.
+    """
+
+    def __init__(self, msg=None):
+        if msg is None:
+            msg = ("An input array is nearly constant; the computed "
+                   "correlation coefficent may be inaccurate.")
+        self.args = (msg,)
+
+
 def pearsonr(x, y):
     r"""
-    Calculate a Pearson correlation coefficient and the p-value for testing
-    non-correlation.
+    Pearson correlation coefficient and p-value for testing non-correlation.
 
-    The Pearson correlation coefficient measures the linear relationship
+    The Pearson correlation coefficient [1]_ measures the linear relationship
```

Comment:
One could replace "Strictly speaking, Pearson's correlation requires that each dataset be normally distributed, and not necessarily zero-mean. " by sth like "The calculation of the p-value relies on the assumption that each dataset is normally distributed."

Reply:
:+1: I'll make a change like that.

Reply:
I see this fixed in the diff -- marking as resolved